src/RokuDeploy.ts

```diff
@@ -652,15 +653,54 @@ export class RokuDeploy {
             throw new errors.UnparsableDeviceResponseError('Invalid response', results);
         }
 
+        this.verboseLog(results.body);
+
         if (results.response.statusCode === 401) {
             throw new errors.UnauthorizedDeviceResponseError('Unauthorized. Please verify username and password for target Roku.', results);
         }
 
+        let rokuMessages = this.getRokuMessagesFromResponseBody(results.body);
+
+        if (rokuMessages.errors.length > 0) {
+            throw new errors.FailedDeviceResponseError(rokuMessages.errors[0], rokuMessages);
+        }
+
         if (results.response.statusCode !== 200) {
-            throw new errors.InvalidDeviceResponseCodeError('Invalid response code: ' + results.response.statusCode);
+            throw new errors.InvalidDeviceResponseCodeError('Invalid response code: ' + results.response.statusCode, results);
         }
     }
 
+    public getRokuMessagesFromResponseBody(body: string): { errors: Array<string>; infos: Array<string>; successes: Array<string> } {
+        let errors = [];
+        let infos = [];
+        let successes = [];
+        let errorRegex = /Shell\.create\('Roku\.Message'\)\.trigger\('[\w\s]+',\s+'(\w+)'\)\.trigger\('[\w\s]+',\s+'(.*?)'\)/igm;
+        let match;
+
+        // eslint-disable-next-line no-cond-assign
+        while (match = errorRegex.exec(body)) {
+            let [fullMatch, messageType, message] = match;
```

Comment:
If you aren't going to use `fullMatch`, you can just use an empty comma. Like this:

```typescript
let [, messageType, message] = match;
```